drivers/dma/dma_si32.c

```diff
@@ -141,6 +213,14 @@ static int dma_si32_config(const struct device *dev, uint32_t channel, struct dm
 		return -EINVAL;
 	}
 
+	/* Prevent messing up (potentially) ongoing DMA operations and their settings by disabling
+	 * the channel before applying new settings.
+	 */
+	ret = dma_si32_stop(dev, channel);
```

Comment:
If the channel is not stopped this would be an error. See the DMA state diagrams https://docs.zephyrproject.org/latest/hardware/peripherals/dma.html#channel-state-machine-expectations

Let me know if something is unclear.

Reply:
Thanks a lot, feeling silly now. Need to rework this PR.

Reply:
Just pushed an improved version.

Regarding the documentation: It seem to me as if is missing the transition from `Configured` to `Configured` by using `dma_config`. Would create a PR for this if true.